File: examples/jl/example_intro.jl
# This code implements the example in the Clarabel
# "Getting Started" documentation

using Clarabel, SparseArrays

P = sparse([3. 1 -1; 0 4 2; 0 0 5])

q = [1,2,-3.]

# equality constraint
Aeq = [1. 1. -1.]
beq = [1.]

# inequality constraint
Aineq = [0. 1. 0.;
         0. 0. 1.]
bineq = [2.,2.]

# SOC constraint
Asoc = -I(3)*1.
bsoc = [0.,0.,0.]

# Clarabel.jl constraint data
A = sparse([Aeq; Aineq; Asoc])
b = [beq;bineq;bsoc]

cones = [Clarabel.ZeroConeT(1),
         Clarabel.NonnegativeConeT(2),
         Clarabel.SecondOrderConeT(3)]

settings = Clarabel.Settings(verbose = true, time_limit = 5)

solver   = Clarabel.Solver()

Clarabel.setup!(solver, P, q, A, b, cones, settings)

solution = Clarabel.solve!(solver)

File: examples/py/example_intro.py
import clarabel
import numpy as np
from scipy import sparse

# Define problem data
P = sparse.csc_matrix(
        [[ 3., 1., -1.],
         [ 1., 4.,  2.],
         [-1., 2.,  5.]])

P = sparse.triu(P).tocsc()

q = np.array([1.,2.,-3.])

# equality constraint
Aeq = sparse.csc_matrix([1.,1.,-1])
beq = np.array([1.])

# equality constraint
Aineq = sparse.csc_matrix(
        [[0., 1., 0.],
         [0., 0., 1.]])

bineq = np.array([2.,2.])

# SOC constraint
Asoc = -sparse.identity(3)
bsoc = np.array([0.,0.,0.])

# Clarabel constraint data
A = sparse.vstack([Aeq,Aineq,Asoc]).tocsc()
b = np.concatenate([beq,bineq,bsoc])

cones = [clarabel.ZeroConeT(1),
         clarabel.NonnegativeConeT(2),
         clarabel.SecondOrderConeT(3)]

settings = clarabel.DefaultSettings()
settings.verbose = True
settings.time_limit = 5.

solver = clarabel.DefaultSolver(P,q,A,b,cones,settings)

solver.solve()
